js/src/components/query/QueryPage.tsx

```diff
@@ -138,12 +157,19 @@ export const QueryPage = () => {
             onViewOptionsChanged={setViewOptions}
             onCancel={handleCancel}
           >
-            {(props) => (
-              <QueryDiffResultView
-                {...props}
-                onAddToChecklist={addToChecklist}
-              />
-            )}
+            {(props) =>
```

Comment:
Please merge the two components to one. So that the check page can use the single QueryDiffResultView